client/src/components/Workflow/Editor/WorkflowEdges.vue

```diff
@@ -1,50 +1,86 @@
 <script setup lang="ts">
-import RawConnector from "./Connector.vue";
-import TerminalConnector from "./TerminalConnector.vue";
-import { computed } from "vue";
-import { useConnectionStore } from "@/stores/workflowConnectionStore";
+import { computed, ref, toRef, type Ref } from "vue";
+import { useConnectionStore, type Connection, type OutputTerminal } from "@/stores/workflowConnectionStore";
 import { storeToRefs } from "pinia";
 import type { TerminalPosition } from "@/stores/workflowEditorStateStore";
 import type { OutputTerminals } from "./modules/terminals";
+import { useDrawableConnection } from "./composables/drawableConnection";
+import { useAnimationFrame } from "@/composables/sensors/animationFrame";
 
 const props = defineProps<{
     draggingConnection: TerminalPosition | null;
     draggingTerminal: OutputTerminals | null;
+    transform: { x: number; y: number; k: number };
 }>();
 
+const canvas: Ref<HTMLCanvasElement | null> = ref(null);
+const context = computed(() => canvas.value?.getContext("2d") ?? null);
+
 const connectionStore = useConnectionStore();
 const { connections } = storeToRefs(connectionStore);
 
-const dragStyle = computed(() => {
-    // TODO: this isn't quite right, but 6 seems to work, probably because it's the outer strokewidth
-    // sync up with Connector.vue ?
-    const strokeWidth = 6;
-    const dragStyle: { [index: string]: string } = {};
-    if (props.draggingConnection) {
-        dragStyle.left = props.draggingConnection.endX - strokeWidth + "px";
-        dragStyle.top = props.draggingConnection.endY - strokeWidth + "px";
+const drawableConnections = computed(() => {
+    return connections.value.map((c) => useDrawableConnection(c));
+});
+
+const drawableDraggingConnection = computed(() => {
+    if (props.draggingConnection && props.draggingTerminal) {
+        const connection: Connection = {
+            input: {
+                connectorType: "input",
+                name: "draggingInput",
+                stepId: -1,
+            },
+            output: props.draggingTerminal as unknown as OutputTerminal,
+        };
+
+        return useDrawableConnection(connection, toRef(props, "draggingConnection"));
+    } else {
+        return null;
     }
-    return dragStyle;
 });
 
+useAnimationFrame(() => {
+    const ctx = context.value;
+    if (!ctx) {
+        return;
+    }
+
+    ctx.canvas.width = ctx.canvas.offsetWidth;
+    ctx.canvas.height = ctx.canvas.offsetHeight;
+
+    ctx.resetTransform();
+    ctx.clearRect(0, 0, ctx.canvas.width, ctx.canvas.height);
+    ctx.translate(props.transform.x, props.transform.y);
```

Comment:
That is awesome, it'll be done just once for the canvas, right ?

Reply:
On every animation frame, once for the entire canvas.